File: 50dias/python/29.py
def LerTextoUsuario():
    texto = input('Digite um texto: ')
    return texto

def EhVogal(letra):
    return letra in ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']

def CensuraTexto(texto):
    textoCensurado = ""
    vogais = ""
    for i in range(len(texto)):
        if EhVogal(texto[i]):
            textoCensurado += '*'
            vogais += texto[i]
        else:
            textoCensurado += texto[i]
    return textoCensurado, vogais

def main():
    texto = LerTextoUsuario()

    textoCensurado, vogais = CensuraTexto(texto)
    print('Texto censurado: ' + textoCensurado)
    print('Vogais: ' + vogais)


main()

File: 50dias/cpp/29.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <string>

using namespace std;

const vector<char> vogais = { 'a', 'e', 'i', 'o', 'u',
                             'A', 'E', 'I', 'O', 'U' };

string LerTextoUsuario()
{
    cout << "Digite um texto: ";
    string texto;
    getline(cin, texto);
    return texto;
}

bool EhVogal(const char& letra)
{
    auto it = find(vogais.begin(), vogais.end(), letra);
    bool encontrou = it != vogais.end();
    return encontrou;
}

void CensuraTexto(
    const string& texto,
    string& textoCensurado,
    string& vogais
)
{
    for (int i = 0; i < texto.size(); i++) {
        if (EhVogal(texto[i])) {
            textoCensurado += '*';
            vogais += texto[i];
        }
        else {
            textoCensurado += texto[i];
        }
    }
}

int main()
{
    string texto = LerTextoUsuario();

    string textoCensurado, vogais;
    CensuraTexto(texto, textoCensurado, vogais);

    cout << "Texto censurado: " << textoCensurado << endl;
    cout << "Vogais: " << vogais << endl;

    return 0;
}
